classes/PrestaShopCollection.php

```diff
@@ -119,7 +119,7 @@ public function __construct($classname, $id_lang = null)
      * @param string $on
      * @param int $type
      *
-     * @return PrestaShopCollection
+     * @return self|void
```

Comment:
```suggestion
     * @return static|void
```

Reply:
I disagree. For `$this`, it's ok to return `self`.

Reply:
But what if a class extends `PrestaShopCollection` and you use the method `join` of this class? The object returned is of the child class, not `PrestaShopCollection` right?

Reply:
No the inherited class return the self object so it will return the inherited class, not the `PrestaShopCollection` class.

Reply:
It's basically what i'm saying 😄 But `self` is confusing in my opinion as it can be both `PrestaShopCollection` or a child class. `static` on the other side always refer to the class instantiated.

Reply:
@atomiix It's weird because `static` is not defined in https://docs.phpdoc.org/3.0/guide/references/phpdoc/types.html but it's defined in PHPStan : https://phpstan.org/writing-php-code/phpdoc-types :exploding_head:

Reply:
@Progi1984 I think phpdoc is not up-to-date because since PHP 8.0, [you can have `static` as return type](https://php.watch/versions/8.0/static-return-type)

Reply:
Better we can use `$this` and it's accepted by two :)

Reply:
Works for me here as we're actually returning `$this` but for a `clone $this` I don't think it would have been the solution :p